quinn-proto/src/frame.rs

```diff
@@ -902,4 +952,30 @@ mod test {
             ref x => panic!("incorrect frame {x:?}"),
         }
     }
+
+    #[test]
+    fn ack_frequency_coding() {
+        let mut buf = Vec::new();
+        AckFrequency::encode(42, 20, 50_000, 1, &mut buf);
+        let frames = frames(buf);
+        assert_eq!(frames.len(), 1);
+        match &frames[0] {
+            Frame::AckFrequency(ack_frequency) => {
+                assert_eq!(ack_frequency.sequence, 42);
+                assert_eq!(ack_frequency.ack_eliciting_threshold, 20);
+                assert_eq!(ack_frequency.request_max_ack_delay, 50_000);
+                assert_eq!(ack_frequency.reordering_threshold, 1);
```

Comment:
If we `derive(Eq, PartialEq)` for the frame struct, and take `&self` in `encode`, this could be more concise.